cmd/export/export.go

```diff
@@ -0,0 +1,188 @@
+	"github.com/vmware-tanzu/velero/pkg/discovery"
+	"github.com/vmware-tanzu/velero/pkg/features"
+	errorsutil "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+type ExportOptions struct {
+	configFlags *genericclioptions.ConfigFlags
+
+	logger    logrus.FieldLogger
+	ExportDir string
+	Context   string
+	Namespace string
+	genericclioptions.IOStreams
+}
+
+func (o *ExportOptions) Complete(c *cobra.Command, args []string) error {
+	// TODO: @alpatel
+	return nil
+}
+
+func (o *ExportOptions) Validate() error {
+	// TODO: @alpatel
+	return nil
+}
+
+func (o *ExportOptions) Run() error {
+	return o.run()
+}
+
+func NewExportCommand(streams genericclioptions.IOStreams) *cobra.Command {
+	o := &ExportOptions{
+		configFlags: genericclioptions.NewConfigFlags(true),
+
+		IOStreams: streams,
+		logger:    logrus.New(),
+	}
+	cmd := &cobra.Command{
+		Use:   "export",
+		Short: "Export the namespace resources in an output directory",
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	addFlagsForOptions(o, cmd)
+
+	return cmd
+}
+
+func addFlagsForOptions(o *ExportOptions, cmd *cobra.Command) {
+	cmd.Flags().StringVar(&o.ExportDir, "export-dir", "export", "The path where files are to be exported")
+	cmd.Flags().StringVar(&o.Context, "context", "", "The kube context, if empty it will use the current context. If --namespace is set it will take precedence")
+	cmd.Flags().StringVar(&o.Namespace, "namespace", "", "The kube namespace to export.")
+}
+
+func (o *ExportOptions) run() error {
+	log := o.logger
+
+	config := o.configFlags.ToRawKubeConfigLoader()
+	rawConfig, err := config.RawConfig()
+	if err != nil {
+		log.Errorf("error in generating raw config")
+		return err
+	}
+	if o.Context == "" {
+		o.Context = rawConfig.CurrentContext
+	}
+
+	if o.Context == "" {
+		log.Errorf("current kubecontext is empty and not kubecontext is specified")
+		return fmt.Errorf("current kubecontext is empty and not kubecontext is specified")
+	}
+
+	var currentContext *api.Context
+	contextName := ""
+
+	for name, ctx := range rawConfig.Contexts {
+		if name == o.Context {
+			currentContext = ctx
+			contextName = name
+		}
+	}
+
+	if currentContext == nil {
+		log.Errorf("currentContext is nil\n")
+		return err
+	}
+
+	if o.Namespace == "" {
+		log.Debugf("--namespace is empty, defaulting to current context namespace\n")
+		if currentContext.Namespace == "" {
+			log.Errorf("current context %s namespace is empty, exiting\n", contextName)
+			return fmt.Errorf("current context %s namespace is empty, exiting\n", contextName)
+		}
+		o.Namespace = currentContext.Namespace
+	}
+
+	log.Debugf("current context is: %s\n", currentContext.AuthInfo)
```

Comment:
https://pkg.go.dev/k8s.io/client-go@v0.21.1/tools/clientcmd/api#AuthInfo

I think that we maybe want to use say user/namespace/cluster here just so we don't leak passwords/tokens etc.

Reply:
will fix this asap